GX-TimeFrequency/src/minimax_grids.F90

```diff
@@ -0,0 +1,465 @@
+  !! @param[out] omega_weights: weights for imaginary frequency grid points
+  !! @param[out] cosft_wt: weights for tau -> omega cosine transform. cos(w*t) factor is included.
+  !! @param[out] cosft_tw: weights for omega -> tau cosine transform. cos(w*t) factor is included.
+  !! @param[out] sinft_wt: weights for tau -> omega sine transform. sin(w*t) factor is included.
+  !! @param[out] max_errors: Max error for the three kind of transforms (same order as previous args)
+  !! @param[out] cosft_duality_error. Max_{ij} |AB - I| where A and B are the cosft_wt and cosft_tw matrices.
+  !! @param[out] ierr: Exit status
+  subroutine gx_minimax_grid(num_points, e_min, e_max, &
+       tau_points, tau_weights, omega_points, omega_weights, &
+       cosft_wt, cosft_tw, sinft_wt, &
+       max_errors, cosft_duality_error, ierr)
+
+    integer, intent(in)                               :: num_points
+    real(kind=dp), intent(in)                         :: e_min, e_max
+    real(kind=dp), allocatable, dimension(:), &  
+         intent(out)                                  :: tau_points, tau_weights
+    real(kind=dp), allocatable, dimension(:), &
+         intent(out)                                  :: omega_points(:), omega_weights(:)
+    real(kind=dp), allocatable, dimension(:, :), &
+         intent(out)                                  :: cosft_wt(:, :), cosft_tw(:, :), sinft_wt(:, :)
+    real(kind=dp), intent(out)                        :: max_errors(3), cosft_duality_error
+    integer, intent(out)                              :: ierr
+
+    ! Internal variables
+    integer, parameter                                :: cos_t_to_cos_w = 1
+    integer, parameter                                :: cos_w_to_cos_t = 2
+    integer, parameter                                :: sin_t_to_sin_w = 3
+    integer                                           :: i_point, j_point
+    real(kind=dp)                                     :: e_range, scaling
+    real(kind=dp), dimension(:), allocatable          :: x_tw
+    real(kind=dp), dimension(:, :), allocatable       :: mat
+
+    ! Begin work
+    e_range = e_max/e_min
+    ierr = 0
+
+    allocate (x_tw(2*num_points))
+
+    call get_points_weights_omega(num_points, e_range, x_tw, ierr)
+    if (ierr /= 0) return
+
+    allocate (omega_points(num_points))
+    allocate (omega_weights(num_points))
+
+    do i_point = 1, num_points
+       omega_points(i_point) = x_tw(i_point)
+       omega_weights(i_point) = x_tw(i_point + num_points)
+    end do
+
+    ! scale the minimax parameters
+    omega_points(:) = omega_points(:)*e_min
+    omega_weights(:) = omega_weights(:)*e_min
+
+    ! set up the minimax time grid
+    call get_points_weights_tau(num_points, e_range, x_tw, ierr)
+    if (ierr /= 0) return
+
+    ! For RPA we include already a factor of two (see later steps)
+    scaling = 2.0_dp
+
+    allocate (tau_points(num_points))
+    allocate (tau_weights(num_points))
+
+    do i_point = 1, num_points
```

Comment:
Same comments as above - doesn't need to be a loop and scale whilst filling  (which you do here :))